File: src/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int leftPointer = 0;
        int rightPointer = s.size() - 1;

        if (s.size() == 1)
        {
            return true;
        }

        while (leftPointer < rightPointer){
            bool leftIsChar = (s[leftPointer] >= 65 && s[leftPointer] <=90) || (s[leftPointer] >= 97 && s[leftPointer] <=122) || (s[leftPointer] <= 57 && s[leftPointer] >= 48);
            bool rightIsChar = (s[rightPointer] >= 65 && s[rightPointer] <=90) || (s[rightPointer] >= 97 && s[rightPointer] <=122)
            || (s[rightPointer] <= 57 && s[rightPointer] >= 48);

            if (!leftIsChar)
            {
                leftPointer++;
                std::cout << "?";
                continue;
            }

            if (!rightIsChar)
            {
                rightPointer--;
                continue;
            }

            if (tolower(s[leftPointer]) != tolower(s[rightPointer]))
            {
                return false;
            }

            leftPointer++;
            rightPointer--;
        }

        return true;
    }
};

File: src/valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        s = s.ToLower();
        List<char> parsedInput = ParseInput(s);

        int leftPointer = 0;
        int rightPointer = parsedInput.Count - 1;

        while (leftPointer < rightPointer) {
            if (parsedInput[leftPointer] != parsedInput[rightPointer]){
                return false;
            }
            leftPointer++;
            rightPointer--;
        }

        return true;
    }

     public List<char> ParseInput(string s) {
        	s = s.ToLower();
			List<char> result = new List<char>(s);
			for (int i = 0; i < s.Length; i++) {
				if ((s[i] >= 97 && s[i] <= 122)
                || (s[i] >= 48 && s[i] <= 57)) {
					continue;
				}
				result.Remove(s[i]);
			}

			return result;
    }
}
